SimG4CMS/PPS/src/PPSPixelSD.cc

```diff
@@ -0,0 +1,437 @@
+                     << " PV " << currentPV->GetName() << " PVid = " << currentPV->GetCopyNo()
+                     << " MVid = " << currentPV->GetMother()->GetCopyNo() << " Unit " << unitID << "\n"
+                     << " primary " << primaryID << " time slice " << tSliceID << " For Track  "
+                     << theTrack->GetTrackID() << " which is a " << theTrack->GetDefinition()->GetParticleName();
+
+  if (theTrack->GetTrackID() == 1) {
+    LogDebug("PPSSim") << " of energy " << theTrack->GetTotalEnergy();
+  } else {
+    LogDebug("PPSSim") << " daughter of part. " << theTrack->GetParentID();
+  }
+
+  if (theTrack->GetCreatorProcess() != NULL)
+    LogDebug("PPSSim") << theTrack->GetCreatorProcess()->GetProcessName();
+  else
+    LogDebug("PPSSim") << "NO process";
+#endif
+
+  currentHit = new PPSPixelG4Hit;
+  currentHit->setTrackID(primaryID);
+  currentHit->setTimeSlice(tSlice);
+  currentHit->setUnitID(unitID);
+  currentHit->setIncidentEnergy(incidentEnergy);
+
+  currentHit->setPabs(Pabs);
+  currentHit->setTof(Tof);
+  currentHit->setEnergyLoss(Eloss);
+  currentHit->setParticleType(ParticleType);
+  currentHit->setThetaAtEntry(ThetaAtEntry);
+  currentHit->setPhiAtEntry(PhiAtEntry);
+
+  currentHit->setMeanPosition(Posizio);
+  currentHit->setEntryPoint(theEntryPoint);
+  currentHit->setExitPoint(theExitPoint);
+
+  currentHit->setParentId(ParentId);
+  currentHit->setVx(Vx);
+  currentHit->setVy(Vy);
+  currentHit->setVz(Vz);
+
+  UpdateHit();
+
+  StoreHit(currentHit);
+}
+
+void PPSPixelSD::CreateNewHitEvo() {
+  currentHit = new PPSPixelG4Hit;
+  currentHit->setTrackID(primaryID);
+  currentHit->setTimeSlice(tSlice);
+  currentHit->setUnitID(unitID);
+  currentHit->setIncidentEnergy(incidentEnergy);
+
+  currentHit->setPabs(Pabs);
+  currentHit->setTof(Tof);
+  currentHit->setEnergyLoss(Eloss);
+  currentHit->setParticleType(ParticleType);
+  currentHit->setThetaAtEntry(ThetaAtEntry);
+  currentHit->setPhiAtEntry(PhiAtEntry);
+
+  currentHit->setEntryPoint(theEntryPoint);
+  currentHit->setExitPoint(theExitPoint);
+
+  currentHit->setParentId(ParentId);
+  currentHit->setVx(Vx);
+  currentHit->setVy(Vy);
+  currentHit->setVz(Vz);
+
+  G4ThreeVector _PosizioEvo;
+  int flagAcc = 0;
+  _PosizioEvo = PosizioEvo(Posizio, Vx, Vy, Vz, Pabs, flagAcc);
+
+  if (flagAcc == 1) {
+    currentHit->setMeanPosition(_PosizioEvo);
+
+    UpdateHit();
+
+    StoreHit(currentHit);
+  }
+}
+
+G4ThreeVector PPSPixelSD::PosizioEvo(
+    const G4ThreeVector& Pos, double vx, double vy, double vz, double pabs, int& accettanza) {
+  accettanza = 0;
+  //Pos.xyz() in mm
+  G4ThreeVector PosEvo;
+  double ThetaX = atan((Pos.x() - vx) / (Pos.z() - vz));
+  double ThetaY = atan((Pos.y() - vy) / (Pos.z() - vz));
+  double X_at_0 = (vx - ((Pos.x() - vx) / (Pos.z() - vz)) * vz) / 1000.;
+  double Y_at_0 = (vy - ((Pos.y() - vy) / (Pos.z() - vz)) * vz) / 1000.;
+
+  double csi = fabs((7000. - pabs) / 7000.);
```

Comment:
all purely numerical variables should be const
many of these seem like values that should be derived from the geometry rather than hardcoded into a sensitive detector class

Reply:
Removed, as said in the previous comment, this piece of code is not needed anymore.